packages/xstate-store/src/store.ts

```diff
@@ -373,11 +393,35 @@ export function createStoreWithProducer<
   transitions: {
     [K in keyof TEventPayloadMap & string]: (
       context: NoInfer<TContext>,
-      event: { type: K } & TEventPayloadMap[K]
+      event: { type: K } & TEventPayloadMap[K],
+      enqueue: EnqueueObject<TEmitted>
     ) => void;
   }
+): Store<TContext, ExtractEventsFromPayloadMap<TEventPayloadMap>, TEmitted>;
+
+export function createStoreWithProducer<
+  TContext extends StoreContext,
+  TEventPayloadMap extends EventPayloadMap,
+  TEmitted extends EventObject = EventObject
+>(
+  producer: NoInfer<
+    (context: TContext, recipe: (context: TContext) => void) => TContext
+  >,
+  initialContextOrConfig: any,
+  transitions?: any
 ): Store<TContext, ExtractEventsFromPayloadMap<TEventPayloadMap>, TEmitted> {
-  return createStoreCore(initialContext, transitions as any, producer);
+  if (
+    typeof initialContextOrConfig === 'object' &&
+    'context' in initialContextOrConfig &&
+    'on' in initialContextOrConfig
+  ) {
+    return createStoreCore(
+      initialContextOrConfig.context,
+      initialContextOrConfig.on as any,
+      producer
+    );
+  }
+  return createStoreCore(initialContextOrConfig, transitions as any, producer);
```

Comment:
u are already casting it here to `any`, so either this is excessive or the `transitions?: any` is excessive ;p personally, I prefer to cast as far down the road as we can, instead of annotating parameters as `any`